src/cli.ts

```diff
@@ -0,0 +1,283 @@
+      },
+
+      replace: {
+        type: 'string',
+        array: true,
+        description:
+          'Apply a simple find/replace on source file names. This can be used to fix some oddities with paths which appear in the source map generation process. Accepts regular expressions.',
+        implies: 'with',
+      },
+      with: {
+        type: 'string',
+        array: true,
+        description: 'See --replace.',
+        implies: 'replace',
+      },
+    })
+    .group(['json', 'tsv', 'html', 'file'], 'Output:')
+    .group(['replace', 'with'], 'Replace:')
+    .help('h')
+    .alias('h', 'help')
+    .showHelpOnFail(false, 'Specify --help for available options')
+    .wrap(null) // Do not limit line length
+    .parserConfiguration({
+      'boolean-negation': false, // Allow --no-root
+    })
+    .check(argv => {
+      if (argv.replace && argv.with && argv.replace.length !== argv.with.length) {
+        throw new Error('--replace flags must be paired with --with flags');
+      }
+
+      return true;
+    })
+    .parse();
+
+  // Trim extra quotes
+  const quoteRegex = /'/g;
+  argv._ = argv._.map(path => path.replace(quoteRegex, ''));
+
+  return argv;
+}
+
+export function logError(message: string, error?: Error): void {
+  if (error) {
+    console.error(chalk.red(message), error);
+  } else {
+    console.error(chalk.red(message));
+  }
+}
+
+export function logWarn(message: string): void {
+  console.warn(chalk.yellow(message));
+}
+
+export function logInfo(message: string): void {
+  console.log(chalk.green(message));
+}
+
+/**
+ * Create options object for `explore` method
+ */
+function getExploreOptions(argv: Arguments): ExploreOptions {
+  let html = true;
+  if (argv.json || argv.tsv) {
+    html = false;
+  }
+
+  let replaceMap: ReplaceMap | undefined;
+  const replaceItems = argv.replace;
+  const withItems = argv.with;
+
+  if (replaceItems && withItems) {
+    replaceMap = replaceItems.reduce<ReplaceMap>((result, item, index) => {
+      result[item] = withItems[index];
+
+      return result;
+    }, {});
+  }
+
+  return {
+    html,
+    file: argv.file,
+    replaceMap,
+    onlyMapped: argv.onlyMapped,
+    noRoot: argv.noRoot,
+  };
+}
+
+interface JsonResult {
+  results: {
+    bundleName: string;
+    files: FileSizeMap;
+  }[];
+}
+
+function outputJson(result: ExploreResult): void {
+  const jsonResultObject: JsonResult = {
+    results: result.bundles.map(({ bundleName, files }) => ({
+      bundleName,
+      files,
+    })),
+  };
+
+  console.log(JSON.stringify(jsonResultObject, null, '  '));
+}
+
+function outputTsv(result: ExploreResult): void {
+  console.log('Source\tSize');
+
+  result.bundles.forEach((bundle, index) => {
+    if (index > 0) {
+      // Separate bundles by empty line
+      console.log();
+    }
+
+    Object.entries(bundle.files).forEach(([source, size]) => {
```

Comment:
Would it be helpful to sort by source or size here?

Reply:
I think a user can do it by himself in a program that works with tsv. But I can add sorting by size if it can be useful.

Reply:
Added sorting by size